services/storehost/extMgr.go

```diff
@@ -400,34 +515,19 @@ func (xMgr *ExtentManager) IsExtentOpenedForReplication(extentID string) bool {
 	return false
 }
 
-type seqNumSnapshot struct {
-	snapshotTime int64
-	beginSeqNum  int64
-	lastSeqNum   int64
-}
+func (xMgr *ExtentManager) RegisterCallbacks(callbacks ExtentCallbacks) {
 
-// getAllExtentsSeqNumSnapshot returns a snapshot of seqnums for all currently
-// active extents, used by 'queueMonitor'.
-// TODO: we need to update this mechanism a bit .. in order to better handle
-// extents that might come and go in between queueMonitor's queries.
-func (xMgr *ExtentManager) getAllExtentsSeqNumSnapshot() map[string]*seqNumSnapshot {
-
-	xMgr.RLock()
-	defer xMgr.RUnlock()
+	// add to list of callback objects
+	xMgr.callbacks = append(xMgr.callbacks, callbacks)
+}
 
-	snapshot := make(map[string]*seqNumSnapshot, len(xMgr.extents))
+func (xMgr *ExtentManager) CallbackDone(ext *extentContext) {
```

Comment:
see comment above .. on mechanism to have callbacks to hold onto the extent context. this function is called when they are 'done' (and want to dereference) the extent-context.